packages/core/test/acceptance/handlers.acceptance.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/core
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  Application,
+  Handler,
+  httpHandler,
+} from '../..';
+import {expect, sinon} from '@loopback/testlab';
+
+describe('http handler', () => {
+  let app: Application;
+  before(givenApp);
+
+  it('returns a Handler', () => {
+    const handler: Handler = httpHandler(app);
+    expect(handler).to.be.type('function');
+  });
+
+  context('Handler', () => {
+    // tslint:disable-next-line
+    let stub: any;
+    after(() => {
+      stub.restore();
+    });
+
+    it('calls the injected app\'s handler function when invoked', async () => {
+      stub = sinon.stub(app, 'handleHttp', () => true);
+      const handler = httpHandler(app);
+      expect(stub.calledOnce).to.be.false();
+      // tslint:disable-next-line
+      await handler({} as any, {} as any);
```

Comment:
Huh, this is rather evil. What if `httpHandler` does some processing with the request/response objects? Please use `ShotRequest` and `ShotResponse` to provide dummy objects. Maybe even better, use shot's `inject` function that calls the handler for you.

```ts
import {inject} from '@loopback/testlab';
// ...

// a dummy promisify
await new Promise(resolve => inject(handler, { url: '/' }, resolve));
expect(stub.calledOnce).to.be.true();
```

(Ideally, we should improve testlab to export a promisified version of `inject`, probably under a more descriptive name, e.g. `simulateHttpRequest`.)